soccer/joystick/GamepadMessage.hpp

```diff
@@ -0,0 +1,60 @@
+#pragma once
+#include <time.hpp>
+
+namespace joystick {
```

Comment:
Alright, question time: walk me through the process of adding a new joystick type. We would have to add a new type similar to GamepadMessage, right? Why not have a `ControllerMessage` instead of this that is a common interface between all controller types? (For reference I'd kind of like to add a WASD controller at some point in time so you don't have to grab a joystick for super basic tests - might be a good new member project actually).

Reply:
The reason would be that it's impossible to come up with any "common interface" for controller types. When we're actually using ROS, it would be like
                      (GamepadMsg)                                              (IntentMessage)
[ Gamepad ]  ---------------------> [ GamepadControlScheme ] --------------------> [ Something ]

                      (KeyboardMsg)                                              (IntentMessage)
[ Keyboard ]  --------------------> [ KeyboardControlScheme ] --------------------> [ Something ]

Right now our `IntentMessage` would be basically `context_->robot_intents` and `context_->motion_setpoints`. We're not using ROS yet though, and I think it's not worth setting some sort of the above setup right now.

Reply:
Yeah, that's what I was figuring you'd say.

Thanks for the clarification! I think this is fine for now.